### ISSUE TITLE
[Proposal] Scaffold should provide an event handler for clicking status bar

### ISSUE THREAD

**Issue by SuhwanCha** [flutter/flutter#96696]
## Use case

`Scaffold` or `CupertinoScaffold` creates `GestureDetector` on status bar, for scrolls to top.
But in nested scaffold, only `GestureDetector` on top-level Scaffold works, and no matter to override event.
So, if you are using `IndexedStack`, and each item in IndexedStack has its own Scaffold, you can't handle events that fit the activated IndexedStack now, but only run designated events.  (i.e. if each page in IndexedStack is scrollable, tapping status bar triggers everything to scroll top https://github.com/flutter/flutter/issues/85603#issuecomment-907757258)

Structure of example below:
```
Scaffold
  - Scaffold
    - ListView
```

|Case1 |Case 2|
|------|---|
|Top-level Scaffold has AppBar|Top-level Scaffold doesn't have AppBar|
|![image](https://user-images.githubusercontent.com/16171447/149616160-cdf8e180-eeeb-4474-9884-8a4e08c4d42c.png)|![image](https://user-images.githubusercontent.com/16171447/149616218-4df45684-5271-4c21-b583-8ceadaf38f71.png)|
|**Problem**: Only top-level Scaffold's GestureDetector can be triggered, there isn’t GestureDetector located on status bar for nested Scaffold|Problem: There is GestureDetector located on status bar on nested Scaffold, but only top-level Scaffold's GestureDetector is triggered, and it can't be handled|

In Case 2, there are two GestureDetectors, but only the parent's GestureDetector works.

|Parent(ScaffoldState#f3333)|Child(ScaffoldState#d9fee)|
|------|---|
|![image](https://user-images.githubusercontent.com/16171447/149616573-458e491f-4da5-4297-8b2e-c16a69dbf1a8.png)|![image](https://user-images.githubusercontent.com/16171447/149616591-eb144320-96ef-4040-ae09-357568cb470d.png)|

```
Scaffold(... state: ScaffoldState#f3333(tickers: tracking 2 tickers))
```
You can see both parent Scaffold and child Scaffold have each GestureDetector on status bar.
I've added GestureDetector to print own context, and only #f3333 (parent)'s GestureDetector works.

## Proposal

**Provides Status bar tapping event handler on Scaffold and CupertinoScaffold.**

```dart
Scaffold(
      onStatusBarTap: () => ...
);
```

If `onStatusBarTap` is null, the default handler should work (scrolls to top on iOS, nothing on Android)

**Comment by mark8044** [comment]
🙏 👍👍

Desperately need this.

Thank you!

**Comment by jamesblasco** [comment]
I suggested a way to separate this logic from the Scaffold here: 
https://github.com/flutter/flutter/issues/74727

**Comment by SuhwanCha** [comment]
> I suggested a way to separate this logic from the Scaffold here: 
> 
> https://github.com/flutter/flutter/issues/74727

How do you think about implement GestureDetector on MeterialApp or CuppedtinoApp rather than Scaffold?

I agree with you, we should separate logic, it's cumbersome now. But if then, I have no idea how we can handle tapping status bar event.

**Comment by jamesblasco** [comment]
Yeah it would be handled inside MaterialApp(WidgetApp) via a GestureDetector. (Notice that on iOS tapping anywhere in the status bar it's always received as a tap on the Offset.zero) 

I proposed to be controlled via Actions, similar on how shortcuts is implemented. Check the full example in codepen for more details

**Comment by SuhwanCha** [comment]
@jamesblasco 
I found your code. Thank you!
Btw, it has been about a year, but isn’t there any PR for your issue?

**Comment by mark8044** [comment]
> Yeah it would be handled inside MaterialApp(WidgetApp) via a GestureDetector. (Notice that on iOS tapping anywhere in the status bar it's always received as a tap on the Offset.zero)
> 
> I proposed to be controlled via Actions, similar on how shortcuts is implemented. Check the full example in codepen for more details



> @jamesblasco
> I found your code. Thank you!
> Btw, it has been about a year, but isn’t there any PR for your issue?

I'm just amazed that there aren't more people complaining about this problem.  Isn't this considered a core function of iOS app?  I tap up there at least 100 times a day....

Hoping one of you geniuses can crack this one
